fastcrypto-vdf/src/class_group/discriminant.rs

```diff
@@ -59,10 +51,63 @@ impl Parameter for Discriminant {
         }
         // Set the lower three bits to ensure that the prime is 7 mod 8 which makes the discriminant 1 mod 8.
         Self::try_from(
-            hash_prime::hash_prime_default(seed, size_in_bits / 8, &[0, 1, 2, size_in_bits - 1])
+            hash_prime::hash_prime(seed, size_in_bits / 8, &[0, 1, 2, size_in_bits - 1])
                 .to_bigint()
                 .expect("Never fails")
                 .neg(),
         )
     }
 }
+
+impl<'de> Deserialize<'de> for Discriminant {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Self::try_from(crate::class_group::bigint_serde::deserialize(deserializer)?)
+            .map_err(serde::de::Error::custom)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_discriminant() {
+        let discriminant = Discriminant::try_from(-BigInt::from(223)).unwrap();
+        assert_eq!(discriminant.bits(), 8);
+        assert_eq!(discriminant.as_bigint(), &-BigInt::from(223));
+
+        // Invalid modulus
+        let candidate = BigInt::from(-29);
+        assert!(candidate.is_negative());
+        assert!(Discriminant::try_from(candidate).is_err());
+
+        // Invalid sign
+        let candidate = BigInt::from(17);
+        assert!(candidate.mod_floor(&BigInt::from(8)) == BigInt::from(1));
+        assert!(Discriminant::try_from(candidate).is_err());
+    }
+
+    #[test]
+    fn test_discriminant_from_seed() {
+        let seed = [1, 2, 3];
+        let target_size = 512;
```

Comment:
shall we do at least 1024?

Reply:
Also with the new paper we could theoretically support 2K+ sizes, shall we do or not? Do we actually have a default strong one, generated with some nothing up my sleeve properties?

Reply:
It does support larger discriminants, but it takes a long time. I'll update the test to generate a 1024 bit discriminant instead (takes ~9 seconds) and compare it to a test vector from chiavdf.

Reply:
Re a larger discriminant, generating a strong "nothing-up-my-sleeve" is straight-forward to do. 1) We demonstrate that the implementation follows chiavdf for 1024 bits, 2) decide on some future information, say a bitcoin block hash or similar, 3) generate a 3072 bit discriminant with that seed and use that as a fixed discriminant..